### PR TITLE
delete endpoint created

### PR DESCRIPTION
**Pull request by that-Gui**

### CODE REVIEW

File: build/index.js [hunk 1]
```diff
@@ -65,6 +91,25 @@ app.listen(port, () => {
...
 //delete question endpoint section
-app.delete('/questions/delete', (req, res) => {
-    res.send(`server got a delete request`);
-});
+app.delete('/questions/:id', (req, res) => __awaiter(void 0, void 0, void 0, function* () {
+    console.log(parseInt(req.params.id));
+    try {
+        let id = req.params.id;
+        let deleteData = yield fsPromises.readFile(library, 'utf8');
+        let jsondeletedata = JSON.parse(deleteData);
```
Comment (halimahexe) [inline build/index.js:99]: Please can you use camel case for variable names?

File: src/index.ts [hunk 2]
```diff
@@ -119,6 +108,25 @@ app.listen(port, () => {
...
 
 //delete question endpoint section
-app.delete('/questions/delete', (req: Request, res: Response) => {
-	res.send(`server got a delete request`);
+app.delete('/questions/:id', async (req: Request, res: Response) => {
+	console.log(parseInt(req.params.id));
+	try {
+		let id = req.params.id;
+		let deleteData = await fsPromises.readFile(library, 'utf8');
+		let jsondeletedata = JSON.parse(deleteData);
```
Comment (halimahexe) [inline src/index.ts]: Please can you use camelCase for variable names? i.e. `jsonDeleteData`

File: src/index.ts [hunk 3]
```diff
@@ -119,6 +108,25 @@ app.listen(port, () => {
...
+		let qmatch = jsondeletedata.questions.findIndex(
```
Comment (halimahexe) [inline src/index.ts]: and `qMatch` here

File: src/index.ts [hunk 4]
```diff
@@ -119,6 +108,25 @@ app.listen(port, () => {
...
+			(item: any) => item.id === id
+		);
+		if (qmatch) {
+			jsondeletedata.questions.splice(qmatch, 1);
+			let updatedJsonString = JSON.stringify(jsondeletedata);
+			await fsPromises.writeFile(library, updatedJsonString);
+			console.log('the question has been  deleted');
```
Comment (halimahexe) [inline src/index.ts:124]: Should we send a message to the user that the question is deleted?

**Review by halimahexe — COMMENTED** [review]
Added a few small comments for review

File: src/index.ts (hunk 2 above)
Comment (that-Gui) [reply src/index.ts]: corrected

**Review by halimahexe — APPROVED** [review]
LGTM

**Review by Gaajia — APPROVED** [review]

**Pull request merged**